From ad957db8458045bb8e31a800f098816147da5bf6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zolt=C3=A1n=20B=C3=B6sz=C3=B6rm=C3=A9nyi?=
 <zboszor@gmail.com>
Date: Wed, 7 Jan 2026 10:28:45 +0100
Subject: [PATCH 03/18] Use a pre-built cli11 dependency
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Do not install bash completion for hailort.
Remove autocomplete_func_ methods from CLI11 users.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>
Upstream-Status: Inappropriate [oe specific]
---
 hailort/cmake/external/cli11.cmake       | 19 +------------------
 hailort/hailortcli/CMakeLists.txt        |  1 -
 hailort/hailortcli/hailortcli.hpp        |  8 --------
 hailort/hailortcli/run2/run2_command.cpp | 15 ---------------
 4 files changed, 1 insertion(+), 42 deletions(-)

diff --git a/hailort/cmake/external/cli11.cmake b/hailort/cmake/external/cli11.cmake
index 31545f3..07ccdb7 100644
--- a/hailort/cmake/external/cli11.cmake
+++ b/hailort/cmake/external/cli11.cmake
@@ -1,20 +1,3 @@
 cmake_minimum_required(VERSION 3.11.0)
 
-include(FetchContent)
-
-FetchContent_Declare(
-    cli11
-    GIT_REPOSITORY https://github.com/hailo-ai/CLI11.git
-    GIT_TAG ae78ac41cf225706e83f57da45117e3e90d4a5b4 # Version 2.2.0 + hailo completion
-    SOURCE_DIR ${HAILO_EXTERNAL_DIR}/cli11-src
-    SUBBUILD_DIR ${HAILO_EXTERNAL_DIR}/cli11-subbuild
-)
-
-# https://stackoverflow.com/questions/65527126/disable-install-for-fetchcontent
-FetchContent_GetProperties(cli11)
-if(NOT cli11_POPULATED)
-    FetchContent_Populate(cli11)
-    if (NOT HAILO_EXTERNALS_EXCLUDE_TARGETS)
-        add_subdirectory(${cli11_SOURCE_DIR} ${cli11_BINARY_DIR} EXCLUDE_FROM_ALL)
-    endif()
-endif()
+find_package(CLI11 REQUIRED CONFIG)
diff --git a/hailort/hailortcli/CMakeLists.txt b/hailort/hailortcli/CMakeLists.txt
index 8b3c0cf..2c1f6f6 100644
--- a/hailort/hailortcli/CMakeLists.txt
+++ b/hailort/hailortcli/CMakeLists.txt
@@ -93,4 +93,3 @@ install(TARGETS hailortcli
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
    CONFIGURATIONS Release
 )
-cli11_install_completion_file(hailortcli)
diff --git a/hailort/hailortcli/hailortcli.hpp b/hailort/hailortcli/hailortcli.hpp
index f86fd54..8aad074 100644
--- a/hailort/hailortcli/hailortcli.hpp
+++ b/hailort/hailortcli/hailortcli.hpp
@@ -80,14 +80,6 @@ public:
         desc_function_ = [non_hidden_values]() {
             return CLI::detail::generate_map(CLI::detail::smart_deref(non_hidden_values), true);
         };
-
-        autocomplete_func_ = [non_hidden_values](const std::string &) {
-            std::vector<std::string> completions;
-            for (const auto &completion : non_hidden_values) {
-                completions.emplace_back(completion.first);
-            }
-            return completions;
-        };
     }
 
 private:
diff --git a/hailort/hailortcli/run2/run2_command.cpp b/hailort/hailortcli/run2/run2_command.cpp
index 21b69f8..889b46d 100644
--- a/hailort/hailortcli/run2/run2_command.cpp
+++ b/hailort/hailortcli/run2/run2_command.cpp
@@ -50,10 +50,6 @@ VStreamNameValidator::VStreamNameValidator(const CLI::Option *hef_path_option, c
         //TODO: support?
         return std::string();
     };
-    autocomplete_func_ = [hef_path_option, net_group_name_option](const std::string&) {
-        // TODO: remove existing names from prev user input
-        return get_values(hef_path_option->as<std::string>(), net_group_name_option->as<std::string>());
-    };
 }
 
 std::vector<std::string> VStreamNameValidator::get_values(const std::string &hef_path, const std::string &net_group_name)
@@ -97,10 +93,6 @@ StreamNameValidator::StreamNameValidator(const CLI::Option *hef_path_option, con
         //TODO: support?
         return std::string();
     };
-    autocomplete_func_ = [hef_path_option, net_group_name_option](const std::string&) {
-        // TODO: remove existing names from prev user input
-        return get_values(hef_path_option->as<std::string>(), net_group_name_option->as<std::string>());
-    };
 }
 
 std::vector<std::string> StreamNameValidator::get_values(const std::string &hef_path, const std::string &net_group_name)
@@ -251,13 +243,6 @@ NetworkGroupNameValidator::NetworkGroupNameValidator(const CLI::Option *hef_path
         //TODO: support?
         return std::string();
     };
-    autocomplete_func_ = [hef_path_option](const std::string&) -> std::vector<std::string>{
-        auto hef = Hef::create(hef_path_option->as<std::string>());
-        if (!hef.has_value()) {
-            return {};
-        }
-        return hef->get_network_groups_names();
-    };
 }
 
 /** NetworkApp */
-- 
2.55.0

